hydroflow_lang/src/graph/ops/lattice_batch.rs

```diff
@@ -1,161 +1,110 @@
+/// This operator is similar to `defer_signal` in that it batches input and releases it when a signal is given. It is also similar to `lattice_fold` in that it folds the input into a single lattice.
+/// So, `lattice_batch` is a combination of both `defer_signal` and `lattice_fold`. This operator is useful when trying to combine a sequence of `defer_signal` and `lattice_fold` operators without unnecessary memory consumption.
 ///
-/// Given a [`Stream`](https://docs.rs/futures/latest/futures/stream/trait.Stream.html)
-/// created in Rust code, `lattice_batch`
-/// is passed the receive end of the channel and when receiving any element
-/// will pass through all received inputs to the output unchanged.
+/// There are two inputs to `lattice_batch`, they are `input` and `signal`.
+/// `input` is the input data flow. Data that is delivered on this input is collected in order inside of the `lattice_batch` operator.
+/// When anything is sent to `signal` the collected data is released downstream. The entire `signal` input is consumed each tick, so sending 5 things on `signal` will not release inputs on the next 5 consecutive ticks.
 ///
-/// ```rustbook
-///     use hydroflow::lattices::Max;
+/// ```hydroflow
+/// use lattices::set_union::SetUnionHashSet;
+/// use lattices::set_union::SetUnionSingletonSet;
 ///
-///     let (tx, rx) = hydroflow::util::unbounded_channel::<()>();
+/// source_iter([1, 2, 3])
+///     -> map(SetUnionSingletonSet::new_from)
+///     -> [input]batcher;
 ///
-///     let mut df = hydroflow::hydroflow_syntax! {
-///         source_iter(0..5) -> persist()
-///             -> map(|x| Max::new(x))
-///             -> lattice_batch::<Max<usize>>(rx)
-///             -> assert_eq([Max::new(4)]);
-///     };
+/// source_iter([()])
+///     -> [signal]batcher;
 ///
-///     tx.send(()).unwrap();
-///
-///     df.run_available();
+/// batcher = lattice_batch::<SetUnionHashSet<usize>>()
+///     -> assert_eq([SetUnionHashSet::new_from([1, 2, 3])]);
 /// ```
 pub const LATTICE_BATCH: OperatorConstraints = OperatorConstraints {
     name: "lattice_batch",
     categories: &[OperatorCategory::LatticeFold],
     persistence_args: RANGE_0,
     type_args: &(0..=1),
-    hard_range_inn: RANGE_1,
-    soft_range_inn: RANGE_1,
+    hard_range_inn: &(2..=2),
+    soft_range_inn: &(2..=2),
     hard_range_out: RANGE_1,
     soft_range_out: RANGE_1,
-    num_args: 1,
+    num_args: 0,
     is_external_input: false,
-    ports_inn: None,
+    ports_inn: Some(|| PortListSpec::Fixed(parse_quote! { input, signal })),
     ports_out: None,
     properties: FlowProperties {
         deterministic: FlowPropertyVal::Preserve,
         monotonic: FlowPropertyVal::No,
         inconsistency_tainted: false,
     },
-    input_delaytype_fn: |_| None,
+    input_delaytype_fn: |_| Some(DelayType::Stratum),
```

Comment:
Is this needed?

Reply:
I think it is for two reasons.

1. to maintain parity with fold/lattice_fold
2. without the stratum boundary then sending things on [signal] is not well defined? Sometimes you will get everything that is on [input] but you aren't guaranteed to get that?

Reply:
2. makes sense to me